libobs/obs-source.c

```diff
@@ -4165,3 +4165,117 @@ EXPORT void obs_enable_source_type(const char *name, bool enable)
 	else
 		info->output_flags |= OBS_SOURCE_CAP_DISABLED;
 }
+
+void obs_media_play_pause(obs_source_t *source)
+{
+	if (!obs_source_valid(source, "obs_media_play_pause"))
+		return;
+
+	const char *id = obs_source_get_id(source);
+	const struct obs_source_info *info = get_source_info(id);
+
+	if (info)
+		info->media_play_pause(source->context.data);
```

Comment:
The code to get the `id` and then get the `obs_source_info` structure is redundant because the `obs_source_info` structure is stored as a part of the source.  So instead of:

```C
	const char *id = obs_source_get_id(source);
	const struct obs_source_info *info = get_source_info(id);
 	if (info)
		info->media_play_pause(source->context.data);
```

This should be:
```C
 	if (source->info.media_play_pause)
		source->info.media_play_pause(source->context.data);
```

Please fix for the rest of the functions you added as well.